### PR TITLE
Cover Bulk Injection of Properties and ConfigValue instances

### PR DESCRIPTION
**Pull request by Sgitario**
These are two new features in MicroProfile 4.0.
See more in https://microprofile.io/2020/12/23/microprofile-4-0-is-now-available/

### CODE REVIEW

**Comment by kshpak** [comment]
I will check out this fully later, but what I see now is, @ConfigProperties was already used in this scenario in AntagonistConfiguration class. The only thing that is added is ConfigValue and property expressions like "http://${server.host}/endpoint"
At the first glance looks like these two aspects will better to incorporate with the current solution that we have

**Comment by Sgitario** [comment]
> I will check out this fully later, but what I see now is, @ConfigProperties was already used in this scenario in AntagonistConfiguration class. The only thing that is added is ConfigValue and property expressions like "http://${server.host}/endpoint"
> At the first glance looks like these two aspects will better to incorporate with the current solution that we have

Not quite. In this PR, we're using the MicroProfile API and proving we can achieve all the new features in MP 4.0.

File: 022-quarkus-properties-config-all/src/main/resources/application.properties [hunk 1]
```diff
@@ -3,4 +3,20 @@ antagonist.message=Hi, I am Sheldon Plankton
...
+# default properties
+server.url.with.default.found=http://${server.host:default.org}/endpoint
+server.url.with.default.not.found=http://${server.host.not.found:default.org}/endpoint
+server.url.with.default.nested=http://${server.host.not.found:${server.host}}/endpoint
+
+# composed properties
+server.url.composed=http://${server.host}:${server.port}/${server.path}
+
+# no substitution
+server.url.raw=\\${server.host}
```
Comment (kshpak) [inline 022-quarkus-properties-config-all/src/main/resources/application.properties:22]: Please add a comment that these properties target .bulk package. So separation of properties is clear

File: 022-quarkus-properties-config-all/README.md [hunk 2]
```diff
@@ -20,6 +20,9 @@ and contains fully qualified names of implementations
 4.2 To create the custom converter POJO class must be created if you're not overriding any of the
 supported classes (e.x. String, Boolean, Double, etc.)
 
+5. Configuring using injection of Bulk Of Properties into an Scoped Bean.
```
Comment (kshpak) [inline 022-quarkus-properties-config-all/README.md:23]: Append: Introduced in MicroProfile Config 2.0

File: 022-quarkus-properties-config-all/README.md [hunk 3]
```diff
@@ -20,6 +20,9 @@ and contains fully qualified names of implementations
 4.2 To create the custom converter POJO class must be created if you're not overriding any of the
 supported classes (e.x. String, Boolean, Double, etc.)
 
+5. Configuring using injection of Bulk Of Properties into an Scoped Bean. 
+
+6. Injecting a `ConfigValue` instance from MicroProfile Config
```
Comment (kshpak) [inline 022-quarkus-properties-config-all/README.md:25]:
```suggestion
6. Injecting a `ConfigValue` instance. Introduced in MicroProfile Config 2.0
```

**Review by kshpak — CHANGES_REQUESTED** [review]

**Comment by kshpak** [comment]
> > I will check out this fully later, but what I see now is, @ConfigProperties was already used in this scenario in AntagonistConfiguration class. The only thing that is added is ConfigValue and property expressions like "http://${server.host}/endpoint"
> > At the first glance looks like these two aspects will better to incorporate with the current solution that we have
> 
> Not quite. In this PR, we're using the MicroProfile API and proving we can achieve all the new features in MP 4.0.

Ambiguous feelings of this PR. On one side I don't see big added value of bulk properties as they were already used by io.quarkus.arc.config.ConfigPropertie library just achieved with another microprofile config api. On the other side, there is added value of property defaults.

**Comment by rsvoboda** [comment]
MP spec has added value for customers, some care about MP compliance a lot (at least that's my experience)
Spec based approaches are ensuring some level of portability

File: 022-quarkus-properties-config-all/README.md (hunk 2 above)
Comment (Sgitario) [reply 022-quarkus-properties-config-all/README.md:23]: Updated

File: 022-quarkus-properties-config-all/README.md (hunk 3 above)
Comment (Sgitario) [reply 022-quarkus-properties-config-all/README.md:25]: Updated

File: 022-quarkus-properties-config-all/src/main/java/io/quarkus/qe/bulk/InjectBulkOfPropertiesConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package io.quarkus.qe.bulk;
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 022-quarkus-properties-config-all/src/main/java/io/quarkus/qe/bulk/InjectBulkOfPropertiesConfiguration.java:1]: The thing is that the standard package name in Beefy TS is not "quarkus.qe", but "io.quarkus.qe". 
I was going to create a separated PR to move "quarkus.qe" to "io.quarkus.qe".

File: 022-quarkus-properties-config-all/src/main/java/io/quarkus/qe/bulk/InjectBulkOfPropertiesConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+import org.eclipse.microprofile.config.inject.ConfigProperties;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+@ConfigProperties(prefix = "server")
+public class InjectBulkOfPropertiesConfiguration {
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 022-quarkus-properties-config-all/src/main/java/io/quarkus/qe/bulk/InjectBulkOfPropertiesConfiguration.java:7]: +1

File: 022-quarkus-properties-config-all/src/main/java/io/quarkus/qe/bulk/InjectBulkOfPropertiesConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    String url;
+    String host;
+    String path;
+    int port;
+
+    @ConfigProperty(name = "port")
+    int repeatedPort;
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 022-quarkus-properties-config-all/src/main/java/io/quarkus/qe/bulk/InjectBulkOfPropertiesConfiguration.java:14]: Yes, I wanted to be sure that if adding the same property twice but using different fields works fine.

File: 022-quarkus-properties-config-all/src/main/java/io/quarkus/qe/bulk/InjectBulkOfPropertiesConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
+package io.quarkus.qe.bulk;
+
+import org.eclipse.microprofile.config.inject.ConfigProperties;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+@ConfigProperties(prefix = "server")
+public class InjectBulkOfPropertiesConfiguration {
+    String url;
+    String host;
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 022-quarkus-properties-config-all/src/main/java/io/quarkus/qe/bulk/InjectBulkOfPropertiesConfiguration.java:9]: If not added, the app fails saying that the `host` value is in the property but is not mapped in this bean.

File: 022-quarkus-properties-config-all/src/main/resources/application.properties (hunk 1 above)
Comment (Sgitario) [reply 022-quarkus-properties-config-all/src/main/resources/application.properties:22]: As discussed, I prefer keeping the properties agnostic from where will be used now and/or in the future.

**Pull request merged**